tests/components/rdw/test_init.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for the RDW integration."""
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from homeassistant.components.rdw.const import DOMAIN
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+
+async def test_load_unload_config_entry(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_rdw: AsyncMock,
+) -> None:
+    """Test the RDW configuration entry loading/unloading."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+
+    await hass.config_entries.async_unload(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert not hass.data.get(DOMAIN)
```

Comment:
Maybe also assert the state of the config entry after unload.
```suggestion
    assert not hass.data.get(DOMAIN)
    assert mock_config_entry.state is ConfigEntryState.NOT_LOADED
```

Reply:
Good suggestion, done 👍